src/classes/RD2_CancelCommitmentService.cls

```diff
@@ -0,0 +1,365 @@
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2020
+* @group Recurring Donations
+* @description Services related to the cancellation of Elevate commitments
+*/
+public without sharing class RD2_CancelCommitmentService {
+
+    /***
+     * @description Max length of a string to be stored in a picklist field
+     */
+    private static final Integer MAX_PICKLIST_FIELD_LEN = 255;
+
+    /***
+     * @description Maximum number of retry attempts to cancel Elevate commitment
+     */
+    @TestVisible
+    private static Integer maxCancelRetries {
+        get {
+            if (maxCancelRetries == null) {
+                maxCancelRetries = 3;
+            }
+            return maxCancelRetries;
+        }
+        set;
+    }
+
+    /**
+    * @description Sends requests and constructs the response
+    */
+    private UTIL_Http.RequestService requestService {
+        get {
+            if (requestService == null) {
+                requestService = new UTIL_Http.RequestService();
+            }
+            return requestService;
+        }
+        set;
+    }
+
+    /**
+    * @description Sends cancel commitment requests for closed Recurring Donations.
+    * Sets RD closed reason back to user specified reason and insert any error.
+    * @return void
+    */
+    public void cancelCommitments(List<RecordWrapper> closedRecords) {
+        List<npe03__Recurring_Donation__c> rdsToUpdate = new List<npe03__Recurring_Donation__c>();
+        List<RecordWrapper> queuedRecords = new List<RecordWrapper>();
+        RD2_ElevateIntegrationService.Logger logger = new RD2_ElevateIntegrationService.Logger();
+
+        for (RecordWrapper record : closedRecords) {
+            Integer calloutsRemaining = Limits.getLimitCallouts();
+
+            if (calloutsRemaining > 0) {
+                //Cancel the commitment or GET the commitment to verify if already cancelled
+                UTIL_Http.Response response = sendRequest(record);
+
+                //Set Recurring Donation Status Reason and its wrapper data to reflect response.
+                handleResponse(record, response, logger);
+
+                //Add the Recurring Donation to update list if it cannot be retried,
+                //it is already cancelled or the cancel succeeded
+                if (isDone(record)) {
```

Comment:
Instead of having a helper method, Use `record.isDone` or move the helper method into the RecordWrapper class and change the line to `record.isDone()`